mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_WebChromeClient.java

```diff
@@ -55,6 +71,127 @@ public boolean onConsoleMessage(final ConsoleMessage message) {
         return super.onConsoleMessage(message);
     }
 
+    /**
+     * show file chooser for WebView on Android >= 16
+     */
+    public void openFileChooser(ValueCallback<Uri> uploadFile, String acceptType, String capture) {
+        final Intent intent = new Intent(Intent.ACTION_GET_CONTENT);
+        intent.addCategory(Intent.CATEGORY_OPENABLE);
+
+        Log.d("Opening file chooser");
+
+        try {
+            mIitc.startActivityForResult(intent, new FileRequestHandlerBelowLollipop(uploadFile));
+        } catch (final ActivityNotFoundException e) {
+            Toast.makeText(mIitc, "No activity to select a file found." +
+                    "Please install a file browser of your choice!", Toast.LENGTH_LONG).show();
+        }
+    }
+
+    private class FileRequestHandlerBelowLollipop implements IITC_Mobile.ResponseHandler {
+        FileRequestHandlerBelowLollipop(ValueCallback<Uri> filePathCallback) {
+            this.filePathCallback = filePathCallback;
+        }
+
+        ValueCallback<Uri> filePathCallback;
+
+        @Override
+        public void onActivityResult(final int resultCode, final Intent data) {
+            mIitc.deleteResponseHandler(this); // to enable garbage collection
+
+            final Uri uri = data.getData();
+            if (uri != null) {
+                this.filePathCallback.onReceiveValue(uri);
+            }
+        }
+    }
+
+    @RequiresApi(Build.VERSION_CODES.LOLLIPOP)
+    private String[] convertToMimeTypes(String[] acceptTypes) {
+        final Set<String> results = new HashSet<>();
+        final MimeTypeMap mtm = MimeTypeMap.getSingleton();
+        for (final String acceptType : acceptTypes) {
+            String convertedType = "*/*";
+            if (acceptType != null) {
+                if (acceptType.contains("/")) {
+                    convertedType = acceptType;
+                } else if (acceptType.startsWith(".") && mtm.hasExtension(acceptType.substring(1))) {
+                    convertedType = mtm.getMimeTypeFromExtension(acceptType.substring(1));
+                }
+            }
+            results.add(Intent.normalizeMimeType(convertedType));
+        }
+        return results.toArray(new String[]{});
```

Comment:
Minor improvement in logic: return empty array if any element is `*/*`.
```js
        for (final String acceptType : acceptTypes) {
            String convertedType = null;
            if (acceptType != null) {
                if (acceptType.contains("/")) {
                    convertedType = Intent.normalizeMimeType(acceptType);
                } else if (acceptType.startsWith(".") && mtm.hasExtension(acceptType.substring(1))) {
                    convertedType = mtm.getMimeTypeFromExtension(acceptType.substring(1));
                }
            }
            if (convertedType == null) { return new String[0]; }
            results.add(convertedType);
        }
        return results.toArray(new String[results.size()]);
```

Reply:
Emmm, Android Studio suggests using empty array for `toArray`

Reply:
@johnd0e I've update this part

Reply:
Well, I don't know Java best practices, so just do it in your way.